vhdl_lang/src/ast.rs

```diff
@@ -1082,19 +1084,45 @@ pub struct LabeledConcurrentStatement {
 /// LRM 13. Design units and their analysis
 #[derive(PartialEq, Debug, Clone)]
 pub struct LibraryClause {
-    pub name_list: Vec<WithRef<Ident>>,
+    pub library_token: TokenId,
+    pub name_list: IdentList,
+    pub semi_token: TokenId,
+}
+
+/// Represents a token-separated list of some generic type `T`
+#[derive(PartialEq, Debug, Clone)]
+pub struct SeparatedList<T> {
```

Comment:
Do we really need to store the TokenId:s of the separators? Can't they be found by accessing the previous token from the list elements if needed?

Reply:
Yes, if the element in the list all have the same token length. For example, for a `library` clause this works:
```
library ieee, std, work;
```
All tokens can be uniquely addressed using the `library` token. However, for more complex `SeparatedList`s this is no longer possible:
```
use ieee.std_logic_1164.all, std.env.all;
```
The count of tokens is not necessarily the same for all elements in the name-list.
My opinion is that this is an optimization for some later point when all token locations are known. Then, given a list of `Name`s, one might be able to advance by the count of tokens in that name to get the location of the `comma` token.